### PR TITLE
Fix false negatives for query reference validation

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

This PR should fix two issues that could lead to cards with issues not showing up in the query validation tool:

1. We were not picking up issues with table-only references (e.g. `SELECT COUNT(*) FROM unknown_table`)
2. Healthy field references could "mask" issues with other fields on a given query.

While this update should fix the issue, it comes at the cost of far less scalable query. Perhaps there's a smart way to optimize this query still, otherwise we might want to consider denormalizing whether a table or field is active on the corresponding analysis rows.

Update: this now fixes a 3rd issue where we didn't de-duplicate table references properly for MBQL cards.

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/query_reference_validation/api.clj [hunk 1]
```diff
@@ -78,6 +78,19 @@
     {:data (map (comp present add-errors) (t2/hydrate cards [:collection :effective_ancestors] :creator))
      :total (t2/count :model/Card (dissoc card-query :limit :offset))}))
 
+(defn- invalid-cards [sort_column sort_direction collection_id]
```
Comment (crisptrutski) [inline enterprise/backend/src/metabase_enterprise/query_reference_validation/api.clj:81]: Having logic inside the endpoint is annoying for REPL usage 😄

File: src/metabase/models/query_analysis.clj [hunk 2]
```diff
@@ -21,22 +21,39 @@
...
+                  [:= :f.id nil]
+                  [:= :f.active false]]]}]
+
+       [:problematic_query_tables
+        {:select [:qt.card_id]
+         :from [[(t2/table-name :model/QueryTable) :qt]]
+         :left-join [[(t2/table-name :model/Table) :t]
+                     [:= :qt.table_id :t.id]]
+         :where [:or
+                 [:= :qt.table_id nil]
```
Comment (crisptrutski) [inline src/metabase/models/query_analysis.clj:46]:
```suggestion
                 [:= :t.id nil]
```

File: src/metabase/query_analysis.clj [hunk 3]
```diff
@@ -91,8 +91,8 @@
 
 (defn- explicit-references [field-ids]
   (let [field-refs (explicit-field-references field-ids)]
-    {:fields field-refs
-     :tables (distinct (map #(dissoc % :field-id :field) field-refs))}))
+    {:fields (distinct field-refs)
```
Comment (crisptrutski) [inline src/metabase/query_analysis.clj:94]: This one is for good luck, the analyzer's *shouldn't* send back duplicates - but maybe they can.

**Review by sansolovyov — APPROVED** [review]
I think sub-selects should be fine, but still would be nice to profile/`explain analyze` it to understand it better.

**Comment by crisptrutski** [comment]
The plan is to revisit the query more rigorously when we do some stress testing.

**Pull request merged**